src/main/java/lab5/reflections.md

```diff
@@ -0,0 +1,99 @@
+        - Because Java supports upcasting (assigning a subclass object to a superclass reference), we can return instances of HumanPlayer, Linus, or Omola without changing the return type.
+
+
+Ques.2   Why pickNextMove must be an abstract method in Player and not just in Player subclasses. ?
+
+Ans.... Why Did the Error Occur Initially?
+  -->   The error occurred because the pickNextMove method was removed from the Player class and placed into the HumanPlayer subclass.
+        The variable whoseTurn is of type Player (the abstract superclass).
+        This caused a compilation error because Java did not know if whoseTurn (a Player) actually had a pickNextMove() method.
+
+Ans.... Why Does Adding the Abstract Method Fix the Error?
+   -->  By adding an abstract method signature for pickNextMove back into the Player class.
+        Java knows that every object of type Player (whether it's HumanPlayer, Linus, or Omola) must have a pickNextMove method.
+
+
+Ques.3   Why it is not necessary to change Main or TicTacToeGame in order to add new Player types to the game. ?
+Ans....  1). Polymorphism: The Key to Flexibility
+          -  Polymorphism allows us to treat different types of players (HumanPlayer, Linus, Omola, etc.) uniformly by referring to them through their shared superclass (Player).
+          -  Even though whoseTurn is a Player reference, at runtime, Java determines which specific implementation of pickNextMove() to call based on the actual object type (whether it's HumanPlayer, Linus, or Omola).
+          -  This means that when we add a new subclass like Omola, the same call (whoseTurn.pickNextMove(board)) will automatically work without any changes to TicTacToeGame.
+
+   2). Abstraction: Hiding Implementation Details
+ - The Player class defines a common interface for all player types by declaring pickNextMove() as an abstract method.
+ - Each specific Player subclass implements this method in its own way (e.g., human input for HumanPlayer, AI logic for Linus and Omola).
+ - The game logic in TicTacToeGame doesn't need to know how a player picks a move—it just calls pickNextMove() and gets a valid move.
+
+3). Encapsulation: Keeping Player Logic Separate
+      - Each Player subclass encapsulates its own logic for picking a move, meaning:
+        ( HumanPlayer prompts the user for input. )
+        ( Linus follows a predefined AI strategy. )
+        ( Omola may use a different algorithm. )
+
+4). The Role of Console.promptForPlayer()
+ - The only minor change required when adding new Player types is inside Console.promptForPlayer(), where we instantiate the correct Player based on user input:
+
+   public static Player promptForPlayer(PlayerToken token) {
+   System.out.print("Enter 'H' for human player, 'L' for Linus, or 'O' for Omola: ");
+   String choice = scanner.nextLine().trim().toUpperCase();
+
+   switch (choice) {
+   case "L":
+   return new Linus();
+   case "O":
```

Comment:
The length and format of this answer makes me think you used AI to generate it...